ui/docs/reference/core/worker-manager/worker-interaction.mdx

```diff
@@ -58,6 +58,19 @@ The expiration time of the new credentials comes from the worker pool configurat
 which gets set when calling [`workerManager.registerWorker`](/docs/reference/core/worker-manager/api#registerWorker).
 The default behavior, when `reregistrationTimeout` is omitted, is to extend the credentials by 96 hours.
 
+## Claim Work Timeout
+
+To make sure that spawned workers are not stuck in a "zombie" state, the worker-manager will check with queue service
+to see if there was any recent activity from a given worker.
+
+Unfortunately, it is not always possible to handle critical errors gracefully on a worker itself. For example,
+if worker process is being terminated with out of memory error, it is not possible to catch it and report it to the worker-manager.
+Also, when image is not configured properly, worker might be able to start and register, but would fail to call `queue.claimWork`.
```

Comment:
```suggestion
Also, when an image is not configured properly, the worker might be able to start and register, but would fail to call `queue.claimWork`.
```